test/test_dynamo_cudagraphs.py

```diff
@@ -0,0 +1,146 @@
+import torch
+from torch.utils._pytree import tree_map
+from torch.fx.passes.backends.cudagraphs import partition_cudagraphs
+from torch.testing._internal.common_utils import (
+    TestCase,
+    run_tests,
+)
+try:
+    import torchdynamo
+    TEST_DYNAMO = True
+except ImportError:
+    TEST_DYNAMO = False
+
+TEST_CUDA = torch.cuda.is_available()
+
+if not TEST_CUDA or not TEST_DYNAMO:
+    print('CUDA or dynamo not available, skipping tests', file=sys.stderr)
+    TestCase = object  # noqa: F811
+
+torchdynamo.config.verify_correctness = True
```

Comment:
Move to a decorator (perhaps unittest.mock.patch), import should not have side effects.